torchmetrics/functional/classification/calibration_error.py

```diff
@@ -0,0 +1,130 @@
+            debias_bins = (acc_bin * (acc_bin - 1) * prop_bin) / (prop_bin * accuracies.size()[0] - 1)
+            ce += torch.sum(torch.nan_to_num(debias_bins))  # replace nans with zeros if nothing appeared in a bin
+        ce = torch.sqrt(ce) if ce > 0 else torch.tensor(0)
+    return ce
+
+
+def _ce_update(preds: Tensor, target: Tensor) -> Tuple[FloatTensor, FloatTensor]:
+    _, _, mode = _input_format_classification(preds, target)
+
+    if mode == DataType.BINARY:
+        confidences, accuracies = preds, target
+    elif mode == DataType.MULTICLASS:
+        confidences, predictions = preds.max(dim=1)
+        accuracies = predictions.eq(target)
+    elif mode == DataType.MULTIDIM_MULTICLASS:
+        # reshape tensors
+        # for preds, move the class dimension to the final axis and flatten the rest
+        confidences, predictions = torch.transpose(preds, 1, -1).flatten(0, -2).max(dim=1)
+        # for targets, just flatten the target
+        accuracies = predictions.eq(target.flatten())
+    else:
+        raise ValueError(
+            f"Calibration error is not well-defined for data with size {preds.size()} and targets {target.size()}."
+        )
+    # must be cast to float for ddp allgather to work
+    return confidences.float(), accuracies.float()
+
+
+def calibration_error(preds: Tensor, target: Tensor, n_bins: int = 15, norm: str = "l1") -> Tensor:
+    r"""
+
+        Computes the top-label calibration error as described in `https://arxiv.org/pdf/1909.10155.pdf`.
+
+        Three different norms are implemented, each corresponding to variations on the calibration error metric.
+
+        L1 norm (Expected Calibration Error)
+
+        .. math::
+            \text{ECE} = \frac{1}{N}\sum_i^N \|(p_i - c_i)\|
+
+        Infinity norm (Maximum Calibration Error)
+
+        .. math::
+            \text{RMSCE} =  \max_{i} (p_i - c_i)
+
+        L2 norm (Root Mean Square Calibration Error)
+
+        .. math::
+            \text{MCE} = \frac{1}{N}\sum_i^N (p_i - c_i)^2
+
+        Where :math:p_i is the top-1 prediction accuracy in
+        bin i and :math:c_i is the average confidence of predictions in bin i.
+
+        # NOTE: L2-norm debiasing is not yet supported.
+
+
+        Args:
+            preds (Tensor): Model output probabilities.
+            target (Tensor): Ground-truth target class labels.
+            n_bins (int, optional): Number of bins to use when computing t. Defaults to 15.
+            norm (str, optional): Norm used to compare empirical and expected probability bins.
+                Defaults to "l1", or Expected Calibration Error.
+    """
+    if norm not in ("l1", "l2", "max"):
+        raise ValueError(f"Norm {norm} is not supported. Please select from l1, l2, or max. ")
+
+    confidences, accuracies = _ce_update(preds, target)
+
+    if not isinstance(n_bins, int) or n_bins <= 0:
+        raise ValueError(f"Expected argument `n_bins` to be a int larger than 0 but got {n_bins}")
```

Comment:
please move in front of the `_ce_update` call so all input checking is in the front